hmpps-prisoner-search-indexer/src/main/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/indexer/repository/IncentiveRepository.kt

```diff
@@ -0,0 +1,106 @@
+package uk.gov.justice.digital.hmpps.prisonersearch.indexer.repository
+
+import org.opensearch.OpenSearchStatusException
+import org.opensearch.action.admin.indices.alias.IndicesAliasesRequest
+import org.opensearch.action.admin.indices.alias.get.GetAliasesRequest
+import org.opensearch.action.admin.indices.delete.DeleteIndexRequest
+import org.opensearch.client.RequestOptions
+import org.opensearch.client.RestHighLevelClient
+import org.opensearch.client.core.CountRequest
+import org.opensearch.client.indices.CreateIndexRequest
+import org.opensearch.client.indices.DeleteAliasRequest
+import org.opensearch.client.indices.GetIndexRequest
+import org.slf4j.LoggerFactory
+import org.springframework.data.elasticsearch.core.ElasticsearchOperations
+import org.springframework.data.elasticsearch.core.mapping.IndexCoordinates
+import org.springframework.data.elasticsearch.core.query.IndexQueryBuilder
+import org.springframework.stereotype.Repository
+import uk.gov.justice.digital.hmpps.prisonersearch.common.config.OpenSearchIndexConfiguration
+import uk.gov.justice.digital.hmpps.prisonersearch.common.model.INCENTIVES_INDEX_STATUS_ID
+import uk.gov.justice.digital.hmpps.prisonersearch.common.model.Incentive
+import uk.gov.justice.digital.hmpps.prisonersearch.common.model.SyncIndex
+
```

Comment:
Very similar to PrisonerRepository but handles the Incentive document class, index and GREEN/BLUE states